components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -79,22 +85,23 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
     )
 
     /**
-     * Deserialize the metrics with a lifetime = User that are on disk.
+     * Deserialize the metrics with a particular lifetime that are on disk.
      * This will be called the first time a metric is used or before a snapshot is
      * taken.
      *
-     * @return A [SharedPreferences] reference that will be used to initialize [userLifetimeStorage]
+     * @param lifetime a [Lifetime] to deserialize
+     * @return A [SharedPreferences] reference that will be used to initialize [pingLifetimeStorage]
+     *         or [userLifetimeStorage] or null if an invalid lifetime is used.
      */
     @Suppress("TooGenericExceptionCaught")
-    open fun deserializeUserLifetime(): SharedPreferences {
-        val prefs =
-            applicationContext.getSharedPreferences(this.javaClass.canonicalName, Context.MODE_PRIVATE)
+    open fun deserializeLifetime(lifetime: Lifetime): SharedPreferences? {
+        val prefs = getPrefsForLifetime(lifetime)
```

Comment:
If we add `require` above, then this could become:

```
val prefsName = if (lifetime == Lifetime.Ping) { 
  "${this.javaClass.canonicalName}.PingLifetime" 
} else {
  this.javaClass.canonicalName
}
val prefs = applicationContext.getSharedPreferences(prefsName, Context.MODE_PRIVATE)
```

This way you won't need to add a new function and the code would be covered by other tests.

Reply:
I refactored it into a function because detekt didn't like the function complexity.  I can suppress the warning and do away with the extra functions.

Reply:
Ok, as long as you add unit tests for it